File: Coin_gogogo/app/src/main/java/com/example/coin_gogogo/Retrofit/Repository.java
package com.example.coin_gogogo.Retrofit;

import com.example.coin_gogogo.data.Candle_List;
import com.example.coin_gogogo.data.Ticker_Response;
import com.example.coin_gogogo.data.Transaction_List_Response;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Single;

public class Repository {

    private static Repository instance;

    private Repository() {
    }

    // 싱클턴으로 사용
    public static Repository getInstance() {
        if (instance == null)
            instance = new Repository();
        return instance;
    }

    public Observable<Ticker_Response> get_Ticker_obs(String paht){
        return RetrofitFactory
                .createRetrofit("https://api.bithumb.com/")
                .create(ExampleService.class)
                .TICKER_OBS(paht);
    }

    public Observable<Transaction_List_Response> get_Transaction_obs(String path, int count){
        return RetrofitFactory
                .createRetrofit("https://api.bithumb.com/")
                .create(ExampleService.class)
                .TRANSACTION_LIST_OBS(path,count);
    }


    public Single<Ticker_Response> get_Ticker(String paht){
        return RetrofitFactory
                .createRetrofit("https://api.bithumb.com/")
                .create(ExampleService.class)
                .TICKER_SINGLE(paht);
    }

    public Single<Transaction_List_Response> get_Transaction_Single(String path, int count){
        return RetrofitFactory
                .createRetrofit("https://api.bithumb.com/")
                .create(ExampleService.class)
                .TRANSACTION_LIST_SINGLE(path,count);
    }

    public Single<Candle_List> get_CandleList_Single(String path){
        return RetrofitFactory
                .createRetrofit("https://api.bithumb.com/")
                .create(ExampleService.class)
                .CANDLE_LIST_SINGLE(path);
    }

}

File: Coin_kotlin/app/src/main/java/com/example/coin_kotlin/retrofit/Repository.kt
package com.example.coin_kotlin.retrofit

import com.example.coin_kotlin.data.Candle_List
import com.example.coin_kotlin.data.Ticker_Response
import io.reactivex.rxjava3.core.Single

object Repository {

    fun get_CandleList_Single(path:String) : Single<Candle_List>?{
        return RetrofitFactory
            .createRetrofit("https://api.bithumb.com/")
            ?.create(Service::class.java)
            ?.CANDLE_LIST_SINGLE(path)
    }

    fun get_Ticker(path: String): Single<Ticker_Response>?{
        return RetrofitFactory
            .createRetrofit("https://api.bithumb.com/")
            ?.create(Service::class.java)
            ?.TICKER_SINGLE(path)
    }

}